quint/src/simulation.ts

```diff
@@ -129,12 +129,17 @@ export function compileAndRun(
       console.error('No trace recorded')
     }
 
+    const errors = ctx.getRuntimeErrors()
+    if (errors.length > 0) {
+      status = 'error'
+    }
+
     return {
       status: status,
       vars: evaluationState.vars.map(v => v.name),
       states: topFrame.args.map(e => e.toQuintEx(idGen)),
       frames: topFrame.subframes,
-      errors: ctx.getRuntimeErrors(),
+      errors: errors,
```

Comment:
I see: looks like this is the core issue? 

We have not tied construction of `errors` to the correct value for status. Mabye the right fix here would be to define

https://github.com/informalsystems/quint/blob/d4b7b0f713bb8f1f693aa2454127f96c771b45f0/quint/src/simulation.ts#L35

So that it also caries all the data essential to each status? So, e.g., for an error `{kind: 'error', errors: errors}` and remove the `errors` field from

https://github.com/informalsystems/quint/blob/d4b7b0f713bb8f1f693aa2454127f96c771b45f0/quint/src/simulation.ts#L37-L47

This is also something we can consider in a followup.